x/ccv/provider/keeper/proposal.go

```diff
@@ -241,26 +260,29 @@ func (k Keeper) SetPendingConsumerAdditionProp(ctx sdk.Context, clientInfo *type
 }
 
 // GetPendingConsumerAdditionProp retrieves a pending proposal to create a consumer chain client (by spawn time and chain id)
-func (k Keeper) GetPendingConsumerAdditionProp(ctx sdk.Context, spawnTime time.Time, chainID string) types.ConsumerAdditionProposal {
+func (k Keeper) GetPendingConsumerAdditionProp(ctx sdk.Context, spawnTime time.Time,
+	chainID string) (prop types.ConsumerAdditionProposal, found bool) {
 	store := ctx.KVStore(k.storeKey)
 	bz := store.Get(types.PendingCAPKey(spawnTime, chainID))
 	if len(bz) == 0 {
-		return types.ConsumerAdditionProposal{}
+		return prop, false
 	}
-	var clientInfo types.ConsumerAdditionProposal
-	k.cdc.MustUnmarshal(bz, &clientInfo)
+	k.cdc.MustUnmarshal(bz, &prop)
 
-	return clientInfo
+	return prop, true
 }
 
 func (k Keeper) PendingConsumerAdditionPropIterator(ctx sdk.Context) sdk.Iterator {
 	store := ctx.KVStore(k.storeKey)
 	return sdk.KVStorePrefixIterator(store, []byte{types.PendingCAPBytePrefix})
 }
 
-// IteratePendingConsumerAdditionProps iterates over the pending consumer addition proposals to create
-// clients in order and creates the consumer client if the spawn time has passed.
-func (k Keeper) IteratePendingConsumerAdditionProps(ctx sdk.Context) {
+// BeginBlockInit iterates over the pending consumer addition proposals in order, and creates
+// clients for props in which the spawn time has passed. Executed proposals are deleted.
+//
+// See: https://github.com/cosmos/ibc/blob/main/spec/app/ics-028-cross-chain-validation/methods.md#ccv-pcf-bblock-init1
+// Spec tag:[CCV-PCF-BBLOCK-INIT.1]
+func (k Keeper) BeginBlockInit(ctx sdk.Context) {
```

Comment:
Changed these methods to use spec naming, since ```Iterate____``` usually means the method directly uses a store iterator